browser/browsing_data/brave_clear_browsing_data.cc

```diff
@@ -0,0 +1,155 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/browsing_data/brave_clear_browsing_data.h"
+#include "base/run_loop.h"
+#include "base/scoped_observer.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/browsing_data/chrome_browsing_data_remover_delegate.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/profiles/profile_manager.h"
+#include "chrome/common/pref_names.h"
+#include "components/browsing_data/core/pref_names.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/browser/browsing_data_remover.h"
+
+namespace {
+
+using content::BraveClearBrowsingData;
+
+class BrowsingDataRemovalWatcher
+    : public content::BrowsingDataRemover::Observer {
+ public:
+  BrowsingDataRemovalWatcher() : observer_(this) {}
+
+  void ClearBrowsingDataForLoadedProfiles(
+      BraveClearBrowsingData::OnExitTestingCallback* testing_callback);
+
+  // BrowsingDataRemover::Observer implementation.
+  void OnBrowsingDataRemoverDone() override;
+
+ private:
+  bool GetClearBrowsingDataOnExitSettings(const Profile* profile,
+                                          int& remove_mask,
+                                          int& origin_mask);
+  void Wait();
+
+  int num_profiles_to_clear_ = 0;
+  base::RunLoop run_loop_;
+  ScopedObserver<content::BrowsingDataRemover,
```

Comment:
I'm a little confused. The is a subclass of `content::BrowsingDataRemover::Observer`, but also has an internal `ScopedObserver`  for `content::BrowsingDataRemover::Observer`?

Reply:
I am following the patter suggested in the comment here: https://cs.chromium.org/chromium/src/content/public/browser/browsing_data_remover.h?l=41 but delegating keeping track of content::BrowsingDataRemover for each profile to ScopedObserver.